actiontext/test/unit/model_test.rb

```diff
@@ -76,6 +76,28 @@ class ActionText::ModelTest < ActiveSupport::TestCase
     assert_equal blob, embeds.first.blob
   end
 
+  test "saving empty content empties the embeds" do
+    blob = create_file_blob(filename: "racecar.jpg", content_type: "image/jpeg")
+    message = Message.create!(subject: "Greetings", content: ActionText::Content.new("Hello world").append_attachables(blob))
+    assert message.content.embeds.sole
+
+    message.content = ""
+    message.save
+
+    assert_empty message.content.embeds
```

Comment:
Have you considered combining these separate assertions into a single [assert_changes](https://edgeapi.rubyonrails.org/classes/ActiveSupport/Testing/Assertions.html#method-i-assert_changes) call?

```suggestion
    assert_changes -> { message.content.embeds.count }, from: 1, to: 0 do
      message.update!(content: "")
    end
```

Reply:
Thank you! Much cleaner!